File: code/community/1304606470934892553/zero-duration-live-transcriptions.go
package main

import (
    "fmt"
    "os"

    "github.com/deepgram/deepgram-sdk-go/deepgram"
)

func main() {
    apiKey := os.Getenv("DEEPGRAM_API_KEY")
    dgClient := deepgram.NewClient(apiKey)

    options := deepgram.TranscriptionOptions{
        Punctuate:   true,
        Encoding:    "mulaw",
        Channels:    2,
        SampleRate:  8000,
        Multichannel: true, // Ensure multichannel is set to true for multiple channels
    }

    response, err := dgClient.StartLiveTranscription(options)
    if err != nil {
        fmt.Printf("Error starting live transcription: %v", err)
        return
    }

    fmt.Printf("Live transcription started successfully: %v", response)

    // For more about multichannel setup, refer to: https://developers.deepgram.com/docs/multichannel
}

File: code/community/1304606470934892553/zero-duration-live-transcriptions.py
import os
from deepgram import Deepgram
import asyncio

async def main():
    api_key = os.getenv('DEEPGRAM_API_KEY')
    deepgram = Deepgram(api_key)

    options = {
        'punctuate': True,
        'encoding': 'mulaw',
        'channels': 2,
        'sample_rate': 8000,
        'multichannel': True  # Ensure multichannel is true for multiple channels
    }

    try:
        response = await deepgram.transcribe_live(options)
        print("Live transcription started successfully:", response)
    except Exception as e:
        print("Error starting live transcription:", e)

# For more about multichannel setup, refer to: https://developers.deepgram.com/docs/multichannel

asyncio.run(main())